### ISSUE TITLE
bindgen does not work with wasm target

### ISSUE THREAD

**Issue by quininer** [quininer/rust-hacl-star#2]
```
> cargo build --target wasm32-unknown-unknown --features bindgen
```

![Screenshot_20190513_111128](https://user-images.githubusercontent.com/6286761/57593847-e082d100-756f-11e9-9b0c-ed14490b0c61.png)

**Comment by quininer** [comment]
This seems to be a bug in clang. https://github.com/rust-lang/rust-bindgen/issues/751

**Comment by raphaelrobert** [comment]
I think what's missing is the standard library for WASM, HACL* has some dependencies (include section in `kremlin_base.h`).

**Comment by quininer** [comment]
I don't think so, but the missing of kremlin does have some impact on this library. Anyway I am planning to migrate to [hacl-star 0.2](https://github.com/project-everest/hacl-star/releases/tag/v.0.2.0).

**Comment by raphaelrobert** [comment]
That's great of course!

Though I'm not sure I understand why libc is not an issue. The dependency is `kremlib.c -> stdlib.h`.

**Comment by raphaelrobert** [comment]
I got it working in the end with the help of https://github.com/jfbastien/musl, but it's not very pretty.

I took a look at EverCrypt and especially the c89 flavor looks really promising, as it has almost no dependency on libc.

**Comment by quininer** [comment]
Ah, I know the problem you have, hacl-star does depend on libc, but this is not what this issue describes (rust-bindgen cannot generate bindings for wasm target).

I haven't followed this issue for a while, but I will migrate it to hacl-star 0.2 next.
